applications/FluidDynamicsApplication/custom_processes/weighted_divergence_calculation_process.h

```diff
@@ -0,0 +1,338 @@
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// This process computes the element average in time of the divergence and of the seminorm of the velocity field.
+/// We define the seminorm of the velocity field as: \left \| \nabla u_{h} \right \|_{L^2(K)}^2 ,
+/// and the divergence as \left \| \nabla \cdot u_{h} \right \|_{L^2(K)}^2 ,
+/// where u is the velocity field and K an element of the domain \Omega.
+/// The time average does not consider the transient 20% first part of the simulation.
+/// The process requires a model part as input.
+
+/** Detail class definition.
+ */
+
+class WeightedDivergenceCalculationProcess: public Process
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of Process
+    KRATOS_CLASS_POINTER_DEFINITION(WeightedDivergenceCalculationProcess);
+
+    typedef ModelPart::ElementType ElementType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor for WeightedDivergenceCalculationProcess Process
+    WeightedDivergenceCalculationProcess(ModelPart& rModelPart
+                    ):
+        Process(),
+        mrModelPart(rModelPart)
+    {
+    }
+
+    /// Destructor.
+    ~WeightedDivergenceCalculationProcess() override {}
+
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /// This operator is provided to call the process as a function and simply calls the Execute method.
+    void operator()()
+    {
+        Execute();
+    }
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /// Execution of the process
+    void Execute() override
+    {
+        KRATOS_TRY;
+
+        // Set time coefficient: computations will be performed ONLY AFTER (time_coefficient * END_TIME)
+        const double time_coefficient = 0.2;
+
+        // Extract time information
+        const ProcessInfo& rCurrentProcessInfo = mrModelPart.GetProcessInfo();
+    	const double& time_step_current  = rCurrentProcessInfo[TIME];
+	    const ProcessInfo& rPreviousProcessInfo = rCurrentProcessInfo.GetPreviousTimeStepInfo();
+        const double& time_step_previous = rPreviousProcessInfo[TIME];
+        const double& final_time = rCurrentProcessInfo[END_TIME];
+
+        if (time_step_current >= time_coefficient * final_time) {
+
+            // Check and set number of elements
+            int number_elements;
+            KRATOS_ERROR_IF(mrModelPart.Elements().size() == 0) << "the number of elements in the domain is zero. weighted divergence calculation cannot be applied"<< std::endl;
+            number_elements = mrModelPart.Elements().size();
+
+            // Check and set number of nodes
+            int number_nodes;
+            KRATOS_ERROR_IF(mrModelPart.Nodes().size() == 0) << "the number of nodes in the domain is zero. weighted divergence calculation cannot be applied" << std::endl;
+            number_nodes = mrModelPart.Nodes().size();
```

Comment:
Where is this one used? I

Reply:
It is not, but I wanted to check that the model part was properly. Now it is removed